src/Tasks/JTask.php

```diff
@@ -69,6 +69,11 @@ public function __construct($params = array())
 		$this->loadConfiguration($params);
 		$this->determineOperatingSystem();
 		$this->determineSourceFolder();
+
+		// Registers the application to run Robo commands
```

Comment:
I can't recall right now, but there are certain scenarios (I think it's when one of those 3 tasks have to run a Robo task) where it fails.  That's why I also moved them up.